### PR TITLE
rgw: share time skew check between v2 and v4 auth

### PR DESCRIPTION
**Pull request by cbodley**
this moves the new std::chrono-based v4 time skew check into a common helper function, then uses that in place of the v2 check

### CODE REVIEW

**Comment by cbodley** [comment]
cc: @qrGitHub

**Comment by qrGitHub** [comment]
LGTM

I will take this out of my todo, thanks for reminding.

File: src/rgw/rgw_auth_s3.cc [hunk 1]
```diff
@@ -223,8 +223,20 @@ namespace rgw {
 namespace auth {
 namespace s3 {
 
-/* FIXME(rzarzynski): duplicated from rgw_rest_s3.h. */
```
Comment (rzarzynski) [inline src/rgw/rgw_auth_s3.cc]: Thank you, @cbodley!

**Comment by cbodley** [comment]
passed qa: http://pulpito.ceph.com/cbodley-2018-01-31_10:32:24-rgw-wip-cbodley-testing-distro-basic-smithi/ with environment failures rerun http://pulpito.ceph.com/cbodley-2018-02-05_09:43:36-rgw-wip-cbodley-testing-distro-basic-smithi/

**Pull request merged**